bool flag
string mode
int32 n
